src/app/tools/point_shapes.h

```diff
@@ -98,20 +98,39 @@ class BrushPointShape : public PointShape {
         color_t b = m_primaryColor;
         const float t = pt.gradient;
         const float ti = 1.0f - pt.gradient;
-        switch (loop->sprite()->pixelFormat()) {
-          case IMAGE_RGB:
-            if (rgba_geta(a) == 0) a = b;
-            else if (rgba_geta(b) == 0) b = a;
+
+        auto rgbaGradient = [&a, b, t, ti] {
+          if (rgba_geta(a) == 0)
+            a = doc::rgba(rgba_getr(b),
+                          rgba_getg(b),
+                          rgba_getb(b),
+                          int(t*rgba_geta(b)));
+          else if (rgba_geta(b) == 0)
+            a = doc::rgba(rgba_getr(a),
+                          rgba_getg(a),
+                          rgba_getb(a),
+                          int(ti*rgba_geta(a)));
+          else
             a = doc::rgba(int(ti*rgba_getr(a) + t*rgba_getr(b)),
                           int(ti*rgba_getg(a) + t*rgba_getg(b)),
                           int(ti*rgba_getb(a) + t*rgba_getb(b)),
                           int(ti*rgba_geta(a) + t*rgba_geta(b)));
+        };
```

Comment:
I'd recommend you to pass a and b as arguments (in the case of indexed, we are using the index b, and not the rgba value of b, because values are captured at this point, and not in the call site, [check this demo](https://godbolt.org/z/vMab7zdbY)). And return the new color:
```suggestion
        auto rgbaGradient = [t, ti](color_t a, color_t b) -> color_t {
          if (rgba_geta(a) == 0)
            return doc::rgba(rgba_getr(b),
                             rgba_getg(b),
                             rgba_getb(b),
                             int(t*rgba_geta(b)));
          else if (rgba_geta(b) == 0)
            return doc::rgba(rgba_getr(a),
                             rgba_getg(a),
                             rgba_getb(a),
                             int(ti*rgba_geta(a)));
          else
            return doc::rgba(int(ti*rgba_getr(a) + t*rgba_getr(b)),
                             int(ti*rgba_getg(a) + t*rgba_getg(b)),
                             int(ti*rgba_getb(a) + t*rgba_getb(b)),
                             int(ti*rgba_geta(a) + t*rgba_geta(b)));
        };
```
Other possibility would be to create a new function it self (passing t and ti too) to avoid confusion with a and b in the parent scope.

Reply:
🤯Very good observation!